perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1029,6 +1033,8 @@ sub process_report : Private {
         $report->anonymous(0);
     } elsif ($c->stash->{contributing_as_anonymous_user}) {
         $report->anonymous(1);
+    } elsif (!$c->stash->{contributing_as_another_user} && $c->user_exists && ($c->user->from_body || $c->user->is_superuser)) {
+        $report->anonymous(0);
     } else {
         $report->anonymous( $params{may_show_name} ? 0 : 1 );
     }
```

Comment:
`contributing_as_another_user` is set in `process_user` which happens after `process_update` we never see the default condition for a staff user.

Reply:
It is set first in process_report, at the top of this function.